re.escape() the query in highlight_content to prevent a server side error.

Solirs opened the pull request:
## What does this PR do?

This PR replaces
`content = re.sub(regex_highlight_cjk(q), f'<span class="highlight">{q}</span>', content)`  
by
`content = re.sub(regex_highlight_cjk(q), f'<span class="highlight">{re.escape(q)}</span>', content)`  
in the highlight_content method to prevent a bad escape error.

## Why is this change important?
When searching for something containing what can look like an escape code (i.e Laminas\XmlRpc\Client) sre_parse can misinterpret it as a bad escape and throw a fatal error preventing the search.

## How to test this PR locally?
make run
Search for "Laminas\XmlRpc\Client"
No error should be thrown and the search should go on normally.


## Author's checklist

N/A

## Related issues
Closes both
#2250 
and
#2256

return42 reviewed (changes requested):
Hi @Solirs thanks a lot for your patch .. the doc of [re.escape](https://docs.python.org/3/library/re.html#re.escape) says:

> This function must not be used for the replacement string in [sub()](https://docs.python.org/3/library/re.html#re.sub) and [subn()](https://docs.python.org/3/library/re.html#re.subn), only backslashes should be escaped.

Here is what I would suggest to fix the issue ..

```diff
diff --git a/searx/webutils.py b/searx/webutils.py
index 7b9a8045c..00929bebe 100644
--- a/searx/webutils.py
+++ b/searx/webutils.py
@@ -174,7 +174,10 @@ def highlight_content(content, query):
             queries.extend(re.findall(regex_highlight_cjk(qs), content, flags=re.I | re.U))
     if len(queries) > 0:
         for q in set(queries):
-            content = re.sub(regex_highlight_cjk(q), f'<span class="highlight">{q}</span>', content)
+            content = re.sub(
+                regex_highlight_cjk(q), f'<span class="highlight">{q}</span>'.replace('\\', r'\\'), content
+            )
     return content
```


To test you can also use ``!qwant ascii \x20``

Solirs committed:
[fix] server side error: escape backslashes in the query highlight_content

Any backslash escapes in the replacement are processed [1], backslashes should
be escaped [2].

[1] https://docs.python.org/3/library/re.html#re.sub
[2] https://docs.python.org/3/library/re.html#re.escape

closes:
- https://github.com/searxng/searxng/issues/2256
- https://github.com/searxng/searxng/issues/2250
